client/signup/steps/plans-pm/index.jsx

```diff
@@ -0,0 +1,193 @@
+import { LoadingEllipsis } from 'calypso/components/loading-ellipsis';
+import MarketingMessage from 'calypso/components/marketing-message';
+import Notice from 'calypso/components/notice';
+import { loadExperimentAssignment } from 'calypso/lib/explat';
+import StepWrapper from 'calypso/signup/step-wrapper';
+import {
+	buildUpgradeFunction,
+	getDomainName,
+	getIntervalType,
+	handleFreePlanButtonClick,
+} from 'calypso/signup/steps/plans/util';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { errorNotice } from 'calypso/state/notices/actions';
+import { saveSignupStep, submitSignupStep } from 'calypso/state/signup/progress/actions';
+import PlansFeaturesMainPM from './plans-features-main-pm';
+import 'calypso/signup/steps/plans/style.scss';
+
+export class PlansStepPM extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			isDesktop: isDesktop(),
+			experiment: null,
+			experimentIsLoading: true,
+		};
+	}
+	componentDidMount() {
+		this.unsubscribe = subscribeIsDesktop( ( matchesDesktop ) =>
+			this.setState( { isDesktop: matchesDesktop } )
+		);
+		this.props.saveSignupStep( { stepName: this.props.stepName } );
+
+		loadExperimentAssignment( 'paid_media_signup_2023_02_hide_monthly' ).then(
+			( experimentName ) => {
+				this.setState( { experiment: experimentName } );
+				this.setState( { experimentIsLoading: false } );
+			}
+		);
+	}
+
+	componentWillUnmount() {
+		this.unsubscribe();
+	}
+
+	onSelectPlan() {
+		return buildUpgradeFunction( this.props );
+	}
+
+	plansFeaturesList() {
+		const { flowName } = this.props;
+
+		let errorDisplay;
+		if ( 'invalid' === this.props.step?.status ) {
+			errorDisplay = (
+				<div>
+					<Notice status="is-error" showDismiss={ false }>
+						{ this.props.step.errors.message }
+					</Notice>
+				</div>
+			);
+		}
+
+		return (
+			<div>
+				{ errorDisplay }
+				<PlansFeaturesMainPM
+					site={ {} }
+					showFAQ={ true }
+					hideFreePlan={ true }
+					intervalType={ getIntervalType() }
+					onUpgradeClick={ this.onSelectPlan() }
+					domainName={ getDomainName( this.props.signupDependencies.domainItem ) }
+					plansWithScroll={ this.state.isDesktop }
+					flowName={ flowName }
+					isAllPaidPlansShown={ true }
+					isInSignup={ true }
+					shouldShowPlansFeatureComparison={ this.state.isDesktop } // Show feature comparison layout in signup flow and desktop resolutions
+					shouldHideMonthlyToggle={ this.state.experiment?.variationName === 'treatment' }
+				/>
+			</div>
+		);
+	}
+
+	getHeaderText() {
+		const { headerText, translate } = this.props;
+
+		if ( headerText ) {
+			return headerText;
+		}
+
+		if ( this.state.isDesktop ) {
+			return translate( 'Choose a plan' );
+		}
+
+		return translate( "Pick a plan that's right for you." );
+	}
+
+	getSubHeaderText() {
+		const { translate } = this.props;
+		const freePlanButton = (
+			<Button onClick={ handleFreePlanButtonClick( this.props ) } borderless />
```

Comment:
This can be changed to 
```suggestion
			<Button onClick={ ()=> buildUpgradeFunction( this.props, null ) } borderless />
```